editor/src/core/shared/multiplayer.ts

```diff
@@ -175,3 +182,34 @@ export function sortThreadsByDescendingUpdateTimeInPlace(
 
   threads.sort((t1, t2) => lastModificationDate(t2).getTime() - lastModificationDate(t1).getTime())
 }
+
+export function useUpdateRemixSceneRouteInLiveblocks() {
+  return useMutation(({ storage, self }, params: { sceneDataLabel: string; location: string }) => {
+    const sceneRoutes = storage.get('remixSceneRoutes')
+    const mySceneRoutes = sceneRoutes.get(self.id)
+    if (mySceneRoutes == null) {
+      const myNewSceneRoutes = new LiveObject({ [params.sceneDataLabel]: params.location })
+      sceneRoutes.set(self.id, myNewSceneRoutes)
+    } else {
+      mySceneRoutes.set(params.sceneDataLabel, params.location)
+    }
+  }, [])
+}
+
+export function useIsOnSameRemixRoute() {
+  const { id } = useSelf()
```

Comment:
it would be best to not use `useSelf` unless you need the presence data, you can instead get the id via `useMyUserId`

Reply:
thanks for the heads up, added it in https://github.com/concrete-utopia/utopia/pull/4744/commits/097d49be5b3746eafe015acea08bb99fad715aef